### PR TITLE
MUMUP-2580 Add To Home With Fname function

### PR DESCRIPTION
**Pull request by Doug-Reed**
Supports change in UW-frame project where one can add an app to home from a feature announcement. 

----

Contributor License Agreement adherence:

- [X ] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: angularjs-portal-home/src/main/webapp/my-app/layout/services.js [hunk 1]
```diff
@@ -27,18 +27,40 @@ define(['angular', 'jquery'], function(angular, $) {
...
-                    $log.warn('failed to add app to home.');
+                    $log.warn('failed to add app to home');
                     return false;
                 },
             });
         };
 
+        var addToHomeWithFname = function addToHomeWithFname(fname) {
+          var tabName = SERVICE_LOC.layoutTab;
+          return $.ajax({
```
Comment (ChristianMurphy) [inline angularjs-portal-home/src/main/webapp/my-app/layout/services.js]: :arrow_right: `$http.post` would be preferable over `$.ajax`

File: angularjs-portal-home/src/main/webapp/my-app/layout/services.js [hunk 2]
```diff
@@ -27,18 +27,40 @@ define(['angular', 'jquery'], function(angular, $) {
                 dataType: 'json',
                 async: true,
                 success: function(request, text) {
-                    $log.log('Added ' + portlet.fname + ' successfully');
+                    $log.log('Added ' + fname + ' successfully');
                     miscService.pushGAEvent(
-                      'Layout Modification', 'Add', portlet.name);
+                      'Layout Modification', 'Add', fname);
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/layout/services.js]: 👍

File: angularjs-portal-home/src/main/webapp/my-app/layout/services.js [hunk 3]
```diff
@@ -27,18 +27,40 @@ define(['angular', 'jquery'], function(angular, $) {
...
                     return true;
                 },
                 error: function(request, text, error) {
-                    $log.warn('failed to add app to home.');
+                    $log.warn('failed to add app to home');
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/layout/services.js]: Would be better if this message said what app, even by fname, failed to add to home.

File: angularjs-portal-home/src/main/webapp/my-app/layout/services.js [hunk 4]
```diff
@@ -27,18 +27,40 @@ define(['angular', 'jquery'], function(angular, $) {
...
+          var tabName = SERVICE_LOC.layoutTab;
+          return $.ajax({
+              url: SERVICE_LOC.base + 'layout?action=addPortlet&fname=' +
+                fname + '&tabName=' + tabName,
+              type: 'POST',
+              data: null,
+              dataType: 'json',
+              async: true,
+              success: function(request, text) {
+                  $log.log('Added ' + fname + ' successfully');
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/layout/services.js]: :+1:

**Pull request closed without merging**